src/name_service/storage/mongo.clj

```diff
@@ -1,4 +1,48 @@
-(ns name-service.storage.mongo)
+(ns name-service.storage.mongo
+  (:require [clojure.spec.alpha :as s]
+            [monger.core :as mg]
+            [monger.collection :as mc]
+            [environ.core :refer [env]]
+            [name-service.core.entity :as entity]
+            [name-service.core.spec :as cs]
+            [clojure.walk :as w]))
 
-(defn make-storage []
-  {})
\ No newline at end of file
+(defn- fetch
+  [{:keys [conn coll-name]} key val]
+  (w/stringify-keys
+    (dissoc
+      (mc/find-one-as-map conn coll-name { key val }) :_id)))
+
+(defn- save
+  [{:keys [conn coll-name] :as storage} keymap key val]
+  (if (s/valid? ::cs/keymap keymap)
+    (w/stringify-keys
+      (dissoc (if (empty? (fetch storage key val))
+                (mc/insert-and-return conn coll-name keymap)
+                (mc/find-and-modify conn coll-name
+                                    {key val}
+                                    keymap
+                                    {:return-new true})
+                ) :_id))
+    :not-valid))
+
+(defn- delete
+  [{:keys [conn coll-name]} key val]
+  (if (and (s/valid? ::cs/key key) (s/valid? ::cs/val val))
+    (mc/remove conn coll-name { key val })
+    :not-valid))
+
+(defrecord MongoStorage [db]
+  entity/KeyBindingStorage
+  (-fetch [_ key val] (fetch db key val))
+  (-save [_ keymap key val] (save db keymap key val))
+  (-delete [_ key val] (delete db key val)))
+
+(defn make-storage [host port db-name coll-name]
+  (let [db {:host host
+            :port (read-string port)
+            :db-name db-name
+            :coll-name coll-name}
+        conn (mg/connect {:host (:host db) :port (:port db)})]
+    (->MongoStorage {:conn (mg/get-db conn (:db-name db))
+                     :coll-name (:coll-name db)})))
```

Comment:
Together with previous changes, we can shorten this:
```suggestion
(defn make-storage
  [{:keys [host port db-name coll-name]}]
  (let [conn (mg/connect {:host host :port (read-string port)})]
    (->MongoStorage {:conn (mg/get-db conn db-name)
                     :coll-name coll-name})))
```

Reply:
Done.